### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3 -L als,ang,ans,dum,geometrie,hist,iff,inout,lod,merget,nd,objext,oder,oposition,parm,parms,pres,poiter,posession,repid,ressource,sinc,som,sur,te,templat,thare,titel,tre`

### CODE REVIEW

File: autotest/ogr/ogr_fgdb.py [hunk 1]
```diff
@@ -2194,7 +2194,7 @@ def test_ogr_fgdb_read_domains():
     domain = ds.GetFieldDomain('SpeedLimit')
     assert domain is not None
     assert domain.GetName() == 'SpeedLimit'
-    assert domain.GetDescription() == 'The maximun speed of the road'
+    assert domain.GetDescription() == 'The maximum speed of the road'
```
Comment (rouault) [inline autotest/ogr/ogr_fgdb.py:2197]: please revert that. the default is in the test dataset

File: autotest/ogr/ogr_openfilegdb.py [hunk 2]
```diff
@@ -1473,7 +1473,7 @@ def test_ogr_openfilegdb_read_domains():
     domain = ds.GetFieldDomain('SpeedLimit')
     assert domain is not None
     assert domain.GetName() == 'SpeedLimit'
-    assert domain.GetDescription() == 'The maximun speed of the road'
+    assert domain.GetDescription() == 'The maximum speed of the road'
```
Comment (rouault) [inline autotest/ogr/ogr_openfilegdb.py:1476]: please revert that. the default is in the test dataset

File: data/grib2_table_4_2_0_2.csv [hunk 3]
```diff
@@ -12,7 +12,7 @@
 6,"MNTSF","Montgomery stream function","(m^2)/(s^2)","UC_NONE"
 7,"SGCVV","Sigma coordinate vertical velocity","1/s","UC_NONE"
 8,"VVEL","Vertical velocity (pressure)","Pa/s","UC_NONE"
-9,"DZDT","Verical velocity (geometric)","m/s","UC_NONE"
+9,"DZDT","Vertical velocity (geometric)","m/s","UC_NONE"
```
Comment (rouault) [inline data/grib2_table_4_2_0_2.csv:15]: the same change should be done in frmts/grib/degrib/data/ which is one of the sources of that generated file

File: data/grib2_table_4_2_10_0.csv [hunk 4]
```diff
@@ -45,7 +45,7 @@
 39,"ALRRC","Altimeter Range Relative Correction","-","UC_NONE"
 40,"MNWSOW","10 Metre Neutral Wind Speed Over Waves","m/s","UC_NONE"
 41,"MWDIRW","10 Metre Wind Direction Over Waves","Degree true","UC_NONE"
-42,"WESP","Wave Engery Spectrum","s/((m^2)*rad)","UC_NONE"
+42,"WESP","Wave Energy Spectrum","s/((m^2)*rad)","UC_NONE"
```
Comment (rouault) [inline data/grib2_table_4_2_10_0.csv:48]: the same change should be done in frmts/grib/degrib/data/ which is one of the sources of that generated file

**Comment by luzpaz** [comment]
@rouault reverted b937643 and added requested changes aaa7d85  
+ added more typo fixes 6b3162c

**Pull request merged**